android-huawei/src/de/golfgl/gdxgamesvcs/HuaweiGameServicesClient.java

```diff
@@ -0,0 +1,631 @@
+import com.huawei.hms.jos.games.archive.OperationResult;
+import com.huawei.hms.jos.games.player.Player;
+import com.huawei.hms.support.api.entity.auth.Scope;
+import com.huawei.hms.support.hwid.HuaweiIdAuthManager;
+import com.huawei.hms.support.hwid.request.HuaweiIdAuthParams;
+import com.huawei.hms.support.hwid.request.HuaweiIdAuthParamsHelper;
+import com.huawei.hms.support.hwid.result.AuthHuaweiId;
+import com.huawei.hms.support.hwid.result.HuaweiIdAuthResult;
+import com.huawei.hms.support.hwid.service.HuaweiIdAuthService;
+
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import de.golfgl.gdxgamesvcs.achievement.IAchievement;
+import de.golfgl.gdxgamesvcs.achievement.IFetchAchievementsResponseListener;
+import de.golfgl.gdxgamesvcs.gamestate.IFetchGameStatesListResponseListener;
+import de.golfgl.gdxgamesvcs.gamestate.ILoadGameStateResponseListener;
+import de.golfgl.gdxgamesvcs.gamestate.ISaveGameStateResponseListener;
+import de.golfgl.gdxgamesvcs.leaderboard.IFetchLeaderBoardEntriesResponseListener;
+import de.golfgl.gdxgamesvcs.leaderboard.ILeaderBoardEntry;
+
+/**
+ * Client for Huawei Game Services
+ * <p>
+ * Created by Francesco Stranieri on 24.07.2020.
+ */
+public class HuaweiGameServicesClient implements IGameServiceClient, AndroidEventListener {
+
+    private final int HUAWEI_GAMESVCS_AUTH_REQUEST = 8971;
+    private final int HUAWEI_GAMESVCS_ACHIEVEMENTS_REQUEST = 8972;
+    private final int HUAWEI_GAMESVCS_LEADERBOARDS_REQUEST = 8973;
+
+    private AndroidApplication activity;
+    private JosAppsClient josAppsClient;
+    private AchievementsClient achievementsClient;
+    private RankingsClient leaderboardsClient;
+    private EventsClient eventsClient;
+    private ArchivesClient archivesClient;
+    private IGameServiceListener gsListener;
+    private Player currentPlayer;
+
+    private boolean isSaveDataEnabled = false;
+    private boolean isSessionActive = false;
+
+    public HuaweiGameServicesClient(AndroidApplication activity, boolean isSaveDataEnabled) {
+        this.activity = activity;
+        this.isSaveDataEnabled = isSaveDataEnabled;
+        this.activity.addAndroidEventListener(this);
+        this.josAppsClient = JosApps.getJosAppsClient(this.activity);
+        this.josAppsClient.init();
+        this.achievementsClient = Games.getAchievementsClient(this.activity);
+        this.leaderboardsClient = Games.getRankingsClient(this.activity);
+        this.eventsClient = Games.getEventsClient(this.activity);
+        this.archivesClient = Games.getArchiveClient(this.activity);
+    }
+
+    @Override
+    public String getGameServiceId() {
+        return IGameServiceClient.GS_HUAWEI_ID;
+    }
+
+    @Override
+    public void setListener(IGameServiceListener gsListener) {
+        this.gsListener = gsListener;
+    }
+
+    @Override
+    public boolean resumeSession() {
+        return true;
```

Comment:
return value javadoc:
true if connection is already established or connection is pending. false if no connect is tried due to unfulfilled prejudices (normally credentials not given).

resume() starts a silent login if possible and connection is not established, logIn() starts a silent login and if this is not possible, a manual login is started